properties/test_index_manipulation.py

```diff
@@ -0,0 +1,159 @@
+import random
+from collections.abc import Hashable
+
+import pytest
+
+from xarray import Dataset
+from xarray.indexes import PandasMultiIndex
+from xarray.testing import _assert_internal_invariants
+
+pytest.importorskip("hypothesis")
+
+import hypothesis.strategies as st
+from hypothesis import assume, note, settings
+from hypothesis.stateful import (
+    Bundle,
+    RuleBasedStateMachine,
+    consumes,
+    invariant,
+    multiple,
+    precondition,
+    rule,
+)
+
+import xarray.testing.strategies as xrst
+
+
+def get_not_multiindex_dims(ds: Dataset) -> tuple[Hashable]:
+    dims = ds.dims
+    mindexes = [
+        name
+        for name, index in ds.xindexes.items()
+        if isinstance(index, PandasMultiIndex)
+    ]
+    return tuple(set(dims) - set(mindexes))
+
+
+def get_multiindex_dims(ds: Dataset) -> list[Hashable]:
+    mindexes = [
+        name
+        for name, index in ds.xindexes.items()
+        if isinstance(index, PandasMultiIndex)
+    ]
+    return mindexes
+
+
+def get_dimension_coordinates(ds: Dataset) -> tuple[Hashable]:
+    return tuple(set(ds.dims) & set(ds._variables))
```

Comment:
Unused code?  Also note that going via `set()` can make tests flaky, via nondeterministic iteration order.